ludwig/models/llm.py

```diff
@@ -72,6 +78,55 @@ def update(self, modules: Dict[str, torch.nn.Module]) -> None:
         self.obj.update(modules)
 
 
+def _should_initialize_model_on_driver(config_obj: LLMModelConfig) -> bool:
+    """Determine if the model should be loaded on the driver or inside training workers.
+
+    This function checks the configuration to decide whether the model should be loaded into memory on the driver
+    or within training workers. If using Ray as the backend and DeepSpeed optimization stage 0, 1, or 2, the model
+    is loaded within the workers since the GPUs need to visible within the RayTrainer process; otherwise, it's
+    loaded at class initialization time.
+
+    Args:
+        config_obj (LLMModelConfig): The language model configuration.
+
+    Returns:
+        bool: True if the model should be loaded on the driver, False otherwise.
+    """
+    backend_type = _get_backend_type_from_config(config_obj)
+    deepspeed_optimization_strategy = _get_deepspeed_optimization_stage_from_config(config_obj)
+    if backend_type == "ray" and deepspeed_optimization_strategy is not None and deepspeed_optimization_strategy <= 2:
+        # If using deepspeed stage 0, 1 or 2, we only load the model into memory once we're actually inside
+        # of the training workers.
+        return False
+    # If using local backend or deepspeed stage 3, we load the model into memory upon class initialization.
+    return True
```

Comment:
Going to push this into the backend class and maybe consider passing in the initialized `self.backend` object in the LudwigModel class into the the create_model function so it gets propagated to here, it shouldn't live here like this.